test/jdk/javax/swing/JFileChooser/NewFolderLocale/FileChooserNewFolderLocaleTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JFileChooser;
+import javax.swing.UIManager;
+
+import java.io.File;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/*
+ * @test
+ * @bug 8312075
+ * @summary Test to check if created newFolder is updated with
+ *          changing Locale.
+ * @run main FileChooserNewFolderLocaleTest
+ */
+
+public class FileChooserNewFolderLocaleTest {
+    static ResourceBundle res;
+    static String newFolderKey;
+    static String newFolderSubKey;
+
+    public static void main(String[] args) throws Exception {
+        File newFolderEnglish = null;
+        File newFolderFrench = null;
+        boolean IS_WINDOWS =
+                System.getProperty("os.name").toLowerCase().contains("windows");
+        if (IS_WINDOWS) {
+            newFolderKey = "FileChooser.win32.newFolder";
+            newFolderSubKey = "FileChooser.win32.newFolder.subsequent";
+        } else {
+            newFolderKey = "FileChooser.other.newFolder";
+            newFolderSubKey = "FileChooser.other.newFolder.subsequent";
+        }
+
+        try {
+            JFileChooser fileChooser = new JFileChooser();
+            File currentDir = new File(".");
+            newFolderEnglish =
+                    fileChooser.getFileSystemView().createNewFolder(currentDir);
+
+            setLocale("fr");
+            fileChooser = new JFileChooser();
+            newFolderFrench =
+                    fileChooser.getFileSystemView().createNewFolder(currentDir);
+            if( !newFolderFrench.getName().contains
+                    (UIManager.getString(newFolderKey))) {
```

Comment:
```suggestion
            if (!newFolderFrench.getName().endsWith(UIManager.getString(newFolderKey))) {
```

Better yet, create a constant for `FRENCH_NEW_FOLDER = "Nouveau dossier"` and use it to put the value to `UIManager` as well as to check it's used.

You may want to ensure, `newFolderEnglish` ends with `"New Folder"`.